src/test/java/test/configuration/BeforeMethodWithGroupFiltersTest.java

```diff
@@ -0,0 +1,18 @@
+package test.configuration;
+
+import org.testng.annotations.Test;
+import test.InvokedMethodNameListener;
+import test.SimpleBaseTest;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class BeforeMethodWithGroupFiltersTest extends SimpleBaseTest {
+
+    @Test
+    public void beforeMethodWithBeforeGroupsShouldOnlyRunBeforeGroupMethods() {
+        InvokedMethodNameListener nameListener = run(new Class[] { BeforeMethodWithGroupFiltersSampleTest.class });
```

Comment:
If `run` has varargs, no need to create the array.
And if `run` doesn't have it, you can modify it.